src/main/java/org/folio/rest/impl/VendorHelper.java

```diff
@@ -63,29 +54,28 @@ public VendorHelper(String lang, HttpClientInterface httpClient, Map<String, Str
   public CompletableFuture<Errors> validateVendor(CompositePurchaseOrder compPO) {
     String id = compPO.getVendor();
     CompletableFuture<Errors> future = new VertxCompletableFuture<>(ctx);
-    List<Error> errors = new ArrayList<>();
     if(id != null) {
       getVendorById(id)
         .thenApply(vendor -> {
           VendorStatus status = VendorStatus.valueOf(vendor.getVendorStatus().toUpperCase());
           if(status != VendorStatus.ACTIVE) {
-            errors.add(createErrorWithId(ORDER_VENDOR_IS_INACTIVE, id));
+            addProcessingError(createErrorWithId(ORDER_VENDOR_IS_INACTIVE, id));
           }
-          return buildErrorsObject(errors);
+          return getProcessingErrors();
```

Comment:
The helper is stateful and shares all the errors already registered prior to this method call. In case the validation of the PO vendor results to success (i.e. no errors found) but the helper already registered errors, for example, for `validateAccessProviders`, the `CompletableFuture` will have `Errors` object with non-empty `Error` list. At least forr me this is confusing...

Reply:
I agree